libbeat/outputs/kafka/client.go

```diff
@@ -254,7 +317,19 @@ func (r *msgRef) dec() {
 	if err != nil {
 		failed := len(r.failed)
 		success := r.total - failed
-		r.batch.RetryEvents(r.failed)
+
+		mr := &msgRef{
+			client:  r.client,
+			count:   int32(len(r.failed)),
+			total:   len(r.failed),
+			failed:  nil,
+			batch:   newRetryBatch(r.failed),
+			retry:   true,
+			backoff: common.NewBackoff(nil, r.backoff.Duration(), 60*time.Second),
+			err:     nil,
+		}
+
+		mr.client.retry(mr)
```

Comment:
This implements infinite retry. Beats must be allowed to drop events.
Calling `r.batch.RetryEvents` will return the failed events within the batch to libbeat. libbeat decides wether to drop or retry.